### ISSUE TITLE
OpenSim_DECLARE_UNNAMED_PROPERTY breaks Visual Studio intellisense

### ISSUE THREAD

**Issue by clnsmith** [opensim-org/opensim-core#2409]
This is a bit of a weird issue... it doesn't cause any errors for compiling etc. But makes developing in Visual Studio more annoying. I am using Visual Studio 15 (2017).

For classes with OpenSim_DECLARE_UNNAMED_PROPERTY in the header, intellisense syntax highlighting doesn't work (ie see ligament.h below)
![image](https://user-images.githubusercontent.com/1234687/52234553-3d605280-28c2-11e9-8e6e-66531215b012.png)

If you comment this line out then the syntax highlighting works:
![image](https://user-images.githubusercontent.com/1234687/52234668-7bf60d00-28c2-11e9-9c45-d2bb52a93136.png)

The reason that this becomes annoying is that then when you try to use ligament in a call to 
ComponentList\<Ligament\> lig_list = _model->updComponentList\<Ligament\>();
you get an intellisense error so that tab completion no longer works. 

![image](https://user-images.githubusercontent.com/1234687/52234935-10606f80-28c3-11e9-8a68-4514a7d2d2f9.png)

The same call with a component that does not use OpenSim_DECLARE_UNNAMED_PROPERTY (ie Force)
ComponentList\<Force\> lig_list = _model->updComponentList\<Force\>();

works with tab completion.

**Comment by clnsmith** [comment]
This might be related more to GeometryPath because ContactGeometry.h and CustomJoint.h both use 
OpenSim_DECLARE_UNNAMED_PROPERTY but don't have the syntax highlighting issues.

**Comment by aymanhab** [comment]
@clnsmith Since I don't see this issue in my environment (VS Community 2017, version 15.9.6) I suspect the issue is specific to your environment. Is there a chance you have customized paths that the wrong GeometryPath is found?

**Comment by clnsmith** [comment]
yes, very possible. I have the 3.3 and 4.0 gui installed and have built opensim from source on this computer. So I am guessing this is the case.    

When you say "paths" do you mean the environmental variable PATH, or some kind of visual studio paths? 

Feel free to close this issue as it appears to be more of a personal one.

**Comment by clnsmith** [comment]
When I used "Go to definition" on GeometryPath in the Ligament.h file, VS would find every reference of GeometryPath in the OpenSim project. This seemed to be localized only to GeometryPath (tried on other components and it went to the real definition). 

I solved the problem by Help->Check for Updates and updating to 15.9.6 (current version). Now everything appears to be working. No idea why this worked. ¯\\_(ツ)_/¯
